specifications/cloud_assembly.md

```diff
@@ -0,0 +1,398 @@
+
+Documents in the archive can be stored with any name and directory structure, however the following entries at the root
+of the archive are reserved for special use:
+* `manifest.json` **MUST** be present and contains the [manifest document](#manifest-document) for the *Cloud Assembly*.
+* `signature.asc`, when present, **MUST** contain the [digital signature](#digital-signature) of the *Cloud Assembly*.
+
+### Manifest Document
+The `manifest.json` file is the entry point of the *Cloud Assembly*. It **MUST** be a valid [JSON] document composed of
+a single `object` that conforms to the following schema:
+
+Key           |Type                 |Required|Description
+--------------|---------------------|:------:|-----------
+`schema`      |`string`             |Required|The schema for the document. **MUST** be `cloud-assembly/1.0`.
+`drops`       |`Map<ID,Drop>`       |Required|A mapping of [*Logical ID*](#logical-id) to [Drop](#drop).
+`missing`     |`Map<string,Missing>`|        |A mapping of context keys to [missing information](#missing).
+
+The [JSON] specification allows for keys to be specified multiple times in a given `object`. However, *Cloud Assembly*
+consumers **MAY** assume keys are unique, and [*Cloud Assemblers*](#cloud-assemblers) **SHOULD** avoid generating
+duplicate keys. If duplicate keys are present, the latest specified value **SHOULD** be preferred.
+
+### Logical ID
+*Logical IDs* are `string`s that uniquely identify [Drop](#drop)s in the context of a *Cloud Assembly*.
+* A *Logical ID* **MUST NOT** be empty.
+* A *Logical ID* **SHOULD NOT** exceed `256` characters.
+* A *Logical ID* **MUST** be composed of only the following ASCII printable characters:
+  + Upper-case letters: `A` (`0x41`) through `Z` (`0x5A`)
+  + Lower-case letters: `a` (`0x61`) through `z` (`0x7A`)
+  + Numeric characters: `0` (`0x30`) through `9` (`0x39`)
+  + Plus: `+` (`0x2B`)
+  + Minus: `-` (`0x2D`)
+  + Forward-slash: `/` (`0x2F`)
+  + Underscore: `_` (`0x5F`)
+* A *Logical ID* **MUST NOT** contain the `.` (`0x2E`) character as it is used in the string substitution pattern for
+  cross-drop references to separate the *Logical ID* from the *attribute* name.
+
+In other words, *Logical IDs* are expected to match the following regular expression:
+```js
+/^[A-Za-z0-9+\/_-]{1,256}$/
+```
+
+### Drop
+Clouds are made of Drops. Thet are the building blocks of *Cloud Assemblies*. They model a part of the
+*cloud application* that can be deployed independently, provided it's dependencies are fulfilled. Drops are specified
+using [JSON] objects that **MUST** conform to the following schema:
+
+Key          |Type             |Required|Description
+-------------|-----------------|:------:|-----------
+`type`       |`string`         |Required|The [*Drop Type*](#drop-type) specifier of this Drop.
+`environment`|`string`         |required|The target [environment](#environment) in which Drop is deployed.
```

Comment:
"is" deployed? "should be" deployed?

Reply:
"should" sounds like a recommendation, where it's a requirement. I can switch to "will be" ?